lib/CustomFields/components/CustomFieldsForm/CustomFieldsForm.js

```diff
@@ -0,0 +1,340 @@
+    dispatchFormAction({ type: 'ADD_NEW_FIELD', payload: { type: fieldType } });
+  };
+
+  const updateCustomField = fieldId => data => {
+    dispatchFormAction({ type: 'UPDATE_FIELD_VALUE', payload: { fieldId, data } });
+  };
+
+  const onDeleteClick = fieldId => () => {
+    dispatchFormAction({ type: 'DELETE_CUSTOM_FIELD', payload: { fieldId } });
+  };
+
+  const resetFormValues = () => {
+    dispatchFormAction({ type: 'RESET_FORM' });
+  };
+
+  const cancelDelete = () => {
+    setDeleteModalIsDisplayed(false);
+    dispatchFormAction({ type: 'CANCEL_DELETE' });
+  };
+
+  const renderFirstMenu = () => {
+    return (
+      <PaneMenu>
+        <PaneHeaderIconButton
+          onClick={redirectToView}
+          icon="times"
+        />
+      </PaneMenu>
+    );
+  };
+
+  const handleExpandAll = (newAccordionStatus) => {
+    setAccordions(newAccordionStatus);
+  };
+
+  const saveCustomFields = () => {
+    onSubmit(customFields.map(cf => ({ ...cf.values })));
+  };
+
+  const onSaveClick = () => {
+    if (fieldsToDelete.length) {
+      setDeleteModalIsDisplayed(true);
+    } else {
+      saveCustomFields();
+    }
+  };
+
+  const renderExpandAllButton = () => (
+    <Row end="xs" className={styles.expandAllWrapper}>
+      <Col xs>
+        <ExpandAllButton
+          accordionStatus={accordions}
+          onToggle={handleExpandAll}
+        />
+      </Col>
+    </Row>
+  );
+
+  const renderPaneFooter = () => {
+    const buttonsDisabled = submitting || (pristine && !fieldsToDelete.length);
+
+    return (
+      <PaneFooter
+        renderStart={
+          <Button
+            marginBottom0
+            onClick={resetFormValues}
+            disabled={buttonsDisabled}
+          >
+            <FormattedMessage id="stripes-smart-components.customFields.cancel" />
+          </Button>
+        }
+        renderEnd={
+          <Button
+            marginBottom0
+            onClick={onSaveClick}
+            disabled={buttonsDisabled}
+          >
+            <FormattedMessage id="stripes-smart-components.customFields.save" />
+          </Button>
+        }
+      />
+    );
+  };
+
+  const onToggleAccordion = ({ id }) => {
+    setAccordions(currentAccordions => ({
+      ...currentAccordions,
+      [id]: !currentAccordions[id],
+    }));
+  };
+
+  const renderFieldAccordions = () => (
+    <div className={styles.accordionsWrapper}>
+      <AccordionSet accordionStatus={accordions} onToggle={onToggleAccordion}>
+        {customFields.map(cf => (
+          <FieldAccordion
+            id={cf.id}
+            formData={cf.values}
+            deleteCustomField={onDeleteClick(cf.id)}
+            editMode
+          >
+            <TextboxFields
+              values={cf.values}
+              onChange={updateCustomField(cf.id)}
+            />
+          </FieldAccordion>
+        ))}
+      </AccordionSet>
+    </div>
+  );
+
+  const renderAccordionName = () => {
+    return (
+      <Fragment>
```

Comment:
There are a few Fragments around single elements in this PR. Is there any particular reason that I'm missing?

(Also just off-hand, we can now use `<>` shorthand rather than having to write out `<Fragment>`)

Reply:
Good catch!

Reply:
As I see, the stripes-core version that is dependent on babel 7 hasn't been released yet